packages/component/src/hooks/useEmojiFromStyles.js

```diff
@@ -0,0 +1,39 @@
+/*eslint no-useless-escape: "off"*/
+/*eslint require-unicode-regexp: "off" */
+import useStyleOptions from './useStyleOptions';
+
+export default function useEmojiFromStyles() {
+  const [{ emojiAutocorrect: autocorrect, emojiList, emojiRegExp: regex }] = useStyleOptions();
+
+  const emojiAutocorrect = autocorrect || false;
+
+  const emojiUnicodeMap = emojiList || {
+    ':)': '😊',
+    ':-)': '😊',
+    '(:': '😊',
+    '(-:': '😊',
+    ':-|': '😐',
+    ':|': '😐',
+    ':-D': '😀',
+    ':D': '😀',
+    ':-p': '😛',
+    ':p': '😛',
+    ':-P': '😛',
+    ':P': '😛',
+    ':-o': '😲',
+    ':o': '😲',
+    ':O': '😲',
+    ':-O': '😲',
+    ':-0': '😲',
+    ':0': '😲',
+    ';-)': '😉',
+    ';)': '😉',
+    '<3': '❤️',
+    '</3': '💔',
+    '<\\3': '💔'
+  };
+
+  const emojiRegExp = regex || new RegExp(/([:<()\\|\/3DPpoO0-]{2,3})/, 'gum');
```

Comment:
I'm not sure I like forcing a developer to write a regex to capture anything that's a key into an arbitrary object they're supplying. You might be able to get away with testing if a substring of the last (maximum length of a key) characters in the textbox is a key in the emojiUnicodeMap object, though I'm not sure how to write that efficiently offhand.

Reply:
I disagree. The regex is clean and precise, and the default regex provided by Web Chat is simple enough that adding or modifying the regex is accessible. It's always going to be up to the user to set up their customization (e.g. the object) correctly.

Reply:
My thought:
- There are two location for same piece of information, in two formats that can't be "naively-converted":
  1. Key of `emojiUnicodeMap`
  1. The "uber" `emojiRegExp`
- Dev can easily say A on (i), and then B on (ii)
   - Things suddenly doesn't work for them
   - To less prone to bugs, we could overcome this by, running all keys against `emojiRegExp` and tell them what need to be added
- Also depends on their `eomjiUnicodeMap`, it could be difficult for them to write a single `RegExp` to capture all of them
- Devs can type `::`. It will be captured in `emojiRegExp`, but the text is not converted to any emojis
   - Then, devs don't understand the value of `emojiRegExp`
   - Then, devs will end up writing `new RegExp(/.*/, 'gum')` and everything still seems to works
   - Then, devs ignore this option

Reply:
Okay, you convinced me. I'll revise this after 4.9.2 release.